src/twisted/web/http.py

```diff
@@ -319,11 +323,26 @@ class _MultiPartParseException(Exception):
     """
 
 
-def _getMultiPartArgs(content: bytes, ctype: bytes) -> dict[bytes, list[bytes]]:
+@dataclass
+class FileUpload:
+    """
+    A file that was uploaded to the web server as part of multipart request.
+    """
+
+    data: BinaryIO
```

Comment:
As an interface, this commits us to a temp-file approach and does not allow for incremental handling of uploads. Could we do a `deliverBody` method or something similar instead?

Reply:
I don't really want to do `deliverBody()` because that will end up being very complex (do producers work? etc).

How about replacing the `data` attribute with a `async get_data(self) -> BinaryIO` function? Because it's async, if `deliverBody` ever gets added `get_data()` can just be refactored to be a utility on top of it. (Ideally I'd want the Rust feature where you consume self but can't do that it in Python, alas.)

Reply:
> I don't really want to do `deliverBody()` because that will end up being very complex (do producers work? etc).

Eh… understandable.

> How about replacing the `data` attribute with a `async get_data(self) -> BinaryIO` function? Because it's async, if `deliverBody` ever gets added `get_data()` can just be refactored to be a utility on top of it. (Ideally I'd want the Rust feature where you consume self but can't do that it in Python, alas.)

Sure, that's baseline acceptable, I think.

Reply:
Wait, what do you mean by "consume self"?

Reply:
So in rust you can say "once I call this method on value X, you can't use X anymore". Which would save the trouble of dealing with "what happens if you call `get_data()` multiple times" edge case, because it would be impossible.

```rust
impl MyStruct {
    /// This involves a read-only reference. Compiler will ensure that while this reference exists, no mutable references will be created for self.
    fn get_info(&self) -> Info { ... }
    
    /// This involves a mutable reference. Compiler will ensure that while this mutable reference exists, no other references, both regular and mutable, can be created for self.
   fn mutate(&mut self) { ... }

   /// This takes ownership of self, "consuming" it. Compiler will ensure no one but us can access it again.
   fn consume(self) -> TransformedMyStruct { ... }
}
```